ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/toolAdmin/view/subviews/ToolContainerEditor.java

```diff
@@ -0,0 +1,129 @@
+package org.iplantc.de.admin.desktop.client.toolAdmin.view.subviews;
+
+import org.iplantc.de.admin.desktop.client.toolAdmin.ToolAdminView;
+import org.iplantc.de.client.models.tool.ToolContainer;
+
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.editor.client.Editor;
+import com.google.gwt.uibinder.client.UiBinder;
+import com.google.gwt.uibinder.client.UiField;
+import com.google.gwt.uibinder.client.UiHandler;
+import com.google.gwt.user.client.ui.HTML;
+import com.google.gwt.user.client.ui.Widget;
+import com.google.inject.Inject;
+
+import com.sencha.gxt.widget.core.client.Composite;
+import com.sencha.gxt.widget.core.client.button.TextButton;
+import com.sencha.gxt.widget.core.client.event.SelectEvent;
+import com.sencha.gxt.widget.core.client.form.FieldLabel;
+import com.sencha.gxt.widget.core.client.form.FieldSet;
+import com.sencha.gxt.widget.core.client.form.IntegerField;
+import com.sencha.gxt.widget.core.client.form.TextField;
+import com.sencha.gxt.widget.core.client.tips.QuickTip;
+
+public class ToolContainerEditor extends Composite implements Editor<ToolContainer> {
+
+    interface ToolContainerEditorBinder extends UiBinder<Widget, ToolContainerEditor> {
+    }
+
+    private static ToolContainerEditorBinder uiBinder = GWT.create(ToolContainerEditorBinder.class);
+
+    @UiField TextField nameEditor;
+    @UiField TextField workingDirectoryEditor;
+    @Ignore
+    @UiField HTML entryPointWarningHTML;
+    @UiField TextField entryPointEditor;
+    @UiField IntegerField memoryLimitEditor;
+    @UiField IntegerField cpuSharesEditor;
```

Comment:
Yeah, it's a little odd, and maybe those JSON keys should actually be made optional in the endpoint, since it doesn't exactly make sense to require an empty array instead of just allowing that key to be omitted.